frontend/src/pages/offers/CreateOfferForm.tsx

```diff
@@ -0,0 +1,118 @@
+
+interface Props {
+  /**
+   * This shipment ID to determine where this offer will be created.
+   */
+  shipmentId: number
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: OfferCreateInput) => void
+}
+
+/**
+ * This form allows users to create a new offer. It is very specific to offer
+ * creation, and therefore difficult to reuse for editing offers.
+ */
+const OfferForm: FunctionComponent<Props> = (props) => {
+  // Each group leader is assigned to a single group
+  // We need to figure out which group that is
+  // TODO: handle the fact that admins can be assigned to any number of groups
+  const profile = useContext(UserProfileContext)
+  const { data: groups } = useAllGroupsMinimalQuery()
+
+  const groupForUser = useMemo(() => {
+    if (groups?.listGroups && profile) {
+      return groups.listGroups.find(
+        (group) =>
+          group.captainId === profile.id &&
+          group.groupType === GroupType.SendingGroup,
+      )
+    }
+  }, [groups, profile])
+
+  const { data: targetShipment } = useShipmentQuery({
+    variables: { id: props.shipmentId },
+  })
+
+  // Set up the form
+  const { register, handleSubmit } = useForm<OfferCreateInput>()
+
+  const onSubmitForm = (input: OfferCreateInput) => {
+    if (!props.shipmentId) {
+      throw new Error('The shipment ID is missing in the offer form')
+    }
+
+    if (!groupForUser) {
+      throw new Error('Cannot find a shipping group in the offer form')
+    }
+
+    // The sending group ID and shipment ID aren't in the form, so we add them
+    // manually
+    input.shipmentId = props.shipmentId
+    input.sendingGroupId = groupForUser.id
+
+    props.onSubmit(input)
+  }
+
+  return (
+    <form onSubmit={handleSubmit(onSubmitForm)} className="space-y-6">
+      <ReadOnlyField label="Sending group">
+        {groupForUser?.name || 'No group'}
+      </ReadOnlyField>
+      <ReadOnlyField label="Shipment">
+        {targetShipment?.shipment && (
+          <>
+            <p className="">{formatShipmentName(targetShipment.shipment)}</p>
+            <p className="text-gray-500 text-sm">
+              {targetShipment.shipment.sendingHub.name} →{' '}
+              {targetShipment.shipment.receivingHub.name}
+            </p>
+          </>
+        )}
+      </ReadOnlyField>
+      <fieldset>
+        <legend>Primary contact</legend>
+        <TextField
+          label="Contact name"
+          name="contact.name"
+          register={register}
+        />
+      </fieldset>
+      <fieldset>
+        <legend>Photos</legend>
+      </fieldset>
+      <Button
+        variant="primary"
+        type="submit"
+        className="mt-6"
+        disabled={props.isLoading}
+      >
+        {props.submitButtonLabel}
```

Comment:
suggestion (non-blocking): since this form is specific to create, maybe it's ok to hard-code this?